src/js/components/Grommet/Grommet.js

```diff
@@ -21,144 +21,90 @@ const FullGlobalStyle = createGlobalStyle`
   body { margin: 0; }
 `;
 
-class Grommet extends Component {
-  static displayName = 'Grommet';
-
-  static getDerivedStateFromProps(nextProps, prevState) {
-    const {
-      background: backgroundProp,
-      dir,
-      theme = {},
-      themeMode,
-    } = nextProps;
-    const { theme: stateTheme, themeProp, themeModeProp } = prevState;
-
-    const nextTheme = deepMerge(baseTheme, theme);
-    if (!stateTheme || theme !== themeProp || themeMode !== themeModeProp) {
-      const {
-        colors: { background: themeBackground },
-      } = nextTheme.global;
-
-      // get initial value for dark so we can normalize background color
-      nextTheme.dark = (themeMode || theme.defaultMode) === 'dark';
-      const color = normalizeColor(
-        backgroundProp || themeBackground,
-        nextTheme,
-      );
-
-      // After normalizing, we set nextTheme.dark once more.
-      // It is necessary that we set it twice. We have to handle two cases:
-      // 1. Caller passes in a color object or a color name that resolves an
-      //    object. In this case, we want to set dark as line 38 shows. The
-      //    second set, in line 46, is a no-op.
-      // 2. Caller passes a specific color value or a color name that resolves
-      //    to a specific color value. In this case, we want dark to be set
-      //    based on that color, which line 46 will do.
-      // The double set of nextTheme.dark allows us to handle both cases here
-      // without having to duplicate color object + name + dark mode detection
-      // code here that is already in normalizeColor and backgroundIsDark.
```

Comment:
Is this comment not necessary anymore or can we include it on the component's refactoring?

Reply:
The comment was a bit hard to figure out since the line numbers were off from some earlier change (a hazard of using line numbers in comments). I think I could probably re-write the description to convey the info with the new hook way of tweaking the theme (it's the same logic and effect.)